### PR TITLE
Add static tests

### PR DESCRIPTION
**Pull request by KellerFuchs**
This uses `flake8` to statically check that the code is “sane-ish”.

Also, this resolves a number of issues that `flake8` found, including superfluous imports.

### CODE REVIEW

**Comment by KellerFuchs** [comment]
Note that I didn't bring the whole thing to Python's *de facto* standard 4-spaces indentation.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+language: python
+python:
+  - 2.7
+
+install:
+  - pip install tox
+
+script:
+  - tox
```
Comment (daurnimator) [inline .travis.yml]: missing newline at EOF

File: provisor/provisor.py [hunk 2]
```diff
@@ -312,33 +306,32 @@ def add_user(self, username, pubkey, hostname,
...
-      'shadowMax' : [ str(nextchange) ],
-      'shadowWarning' : [ str(warning) ],
-      'shadowInactive' : [ str(99999) ],
-      'shadowExpire' : [ str(99999) ],
-      'userPassword' : [ str(password) ],
-      'sshPublicKey' : [ str(pubkey) ],
-      'host' : [ str(hostname) ],
-      'mailRoutingAddress' : [ '{0}@hashbang.sh'.format(username) ],
-      'mailHost' : [ str('smtp:'+hostname) ],
+        'objectClass': [ 'account', 'posixAccount', 'top', 'shadowAccount',
```
Comment (daurnimator) [inline provisor/provisor.py]: I'd split this out to one line per item

**Comment by daurnimator** [comment]
LGTM

**Comment by KellerFuchs** [comment]
@daurnimator Thanks.
Took the liberty to make `master` a protected branch and require that the test pass.

**Pull request merged**